src/api/api.ts

```diff
@@ -269,6 +284,97 @@ export class OpenSeaAPI {
     return assetFromJSON(json);
   }
 
+  /**
+   * Fetch multiple NFTs for a collection from the API
+   * @param slug The collection you would like to list NFTs for
+   * @param limit The number of NFTs to retrieve. Must be greater than 0 and less than 51.
+   * @param next Cursor to retrieve the next page of NFTs
+   * @param retries Number of times to retry if the service is unavailable for any reason
+   */
+  public async getNFTsByCollection(
+    slug: string,
+    limit: number | undefined = undefined,
+    next: string | undefined = undefined,
+    retries = 1
+  ): Promise<ListNFTsResponse> {
+    let response;
+    try {
+      response = await this.get<ListNFTsResponse>(
+        getListNFTsByCollectionPath(slug),
+        {
+          limit,
+          next,
+        }
+      );
+    } catch (error) {
+      _throwOrContinue(error, retries);
+      await delay(1000);
+      return this.getNFTsByCollection(slug, limit, next, retries - 1);
```

Comment:
instead of putting this retry logic everywhere you can possibly just use poll which has expontential backoff logic https://docs.ethers.org/v5/api/utils/web/#utils-poll